students/mayc4t/lesson06/mailroom4_test.py

```diff
@@ -0,0 +1,214 @@
+        self.assertTrue( expected, result)
+        
+        # test donor NOT already in list
+        expected = (False, "New Person" )
+        result = mailroom4.is_donor_in_list("New Person")
+        self.assertTrue( expected, result)
+
+
+    def test_enter_dn_name(self):
+        """
+        Test Enter Donor Name 
+        """
+        def input_mock(self):
+            return 'Michael Kors'
+        mailroom4.input_func = input_mock
+        got_dn_name = mailroom4.enter_dn_name()
+        assert got_dn_name == 'Michael Kors'
+
+
+    def test_enter_dn_gift(self):
+        """
+        Test Enter Gift from donor
+        Only accept input as number
+        or "quit" to back to main menu
+        """
+        def input_mock(self):
+            return 80
+
+        mailroom4.input_func = input_mock
+        got_dn_gift= mailroom4.enter_dn_gift()
+        #enter_dn_gift use input_func
+        #mailroom4.main use input
+        #in mailroom4.main declare 
+        assert got_dn_gift == 80
+    
+
+    def test_add_donation(self):
+        """
+        Test the add_donation 
+        """
+
+        # test add new donor
+        golden_donors_db = {
+            'Lily Maycat': [5000],
+            'Lulu Lemon': [5000, 5000],
+            'Marc Jacobs': [5000, 5000, 5000],
+            'Bobbi Brown': [5000, 5000, 5000, 5000],
+            'Kate Spade': [5000, 5000, 5000, 5000, 5000],
+            'Dragron Four': [300] #new donor to test
+        }
+
+        mailroom4.add_donation("Dragron Four", 300)
+        assert mailroom4.donors_db == golden_donors_db
+
+        # test add the old name + gift
+        golden_donors_db['Kate Spade'].append(300)
+        mailroom4.add_donation( 'Kate Spade', 300)
+        assert mailroom4.donors_db == golden_donors_db
+
+
+    def test_init_db(self):
+        """
+            Make sure the data was created right
+        """
+        #print('test_check_db: %s', mailroom4.donors_db)
+        golden_donors_db = {
+            'Lily Maycat': [5000],
+            'Lulu Lemon': [5000, 5000],
+            'Marc Jacobs': [5000, 5000, 5000],
+            'Bobbi Brown': [5000, 5000, 5000, 5000],
+            'Kate Spade': [5000, 5000, 5000, 5000, 5000],
+        }
+        result = mailroom4.init_db()
+        self.assertTrue(golden_donors_db , result)
+
+
+
+    def test_get_donor_info_by_name(self):
+        """
+            Test get_donor_infor_by_name
+        """
+        #print('test_check_db: %s', mailroom4.donors_db)
+        dn_db = {
+            'Lily Maycat': [5000],
+            'Lulu Lemon': [5000, 5000],
+            'Marc Jacobs': [5000, 5000, 5000],
+            'Bobbi Brown': [5000, 5000, 5000, 5000],
+            'Kate Spade': [5000, 5000, 5000, 5000, 5000],
+        }
+        expected = ['Bobbi Brown', 
+                     sum(dn_db['Bobbi Brown']), 
+                     len(dn_db['Bobbi Brown']),
+                     sum(dn_db['Bobbi Brown'])/len(dn_db['Bobbi Brown'])]
```

Comment:
I'd probably just hard-code the values here since we don't need to test the built ins